coordinator/src/validators/request-validator.ts

```diff
@@ -0,0 +1,80 @@
+import {Bundle, Resource} from "../services/fhir-resources"
+import * as Boom from "@hapi/boom"
+
+function verifyMessageIsResource(message: unknown): message is Resource {
+    return (message as Resource)?.resourceType !== undefined
+}
+
+function verifyResourceTypeIsBundle(resource: unknown): resource is Bundle {
+    return verifyMessageIsResource(resource)
+        && resource.resourceType === "Bundle"
+}
+
+function verifyBundleContainsEntries(bundle: Bundle) {
+    return bundle.entry !== undefined
+}
+
+function getMatchingEntries(bundle: Bundle, resourceType: string) {
+    return bundle.entry
+        .map(entry => entry.resource)
+        .filter(resource => resource.resourceType === resourceType)
+}
+
+function verifyHasId(bundle: Bundle) {
+    return bundle.id !== undefined
+}
+
+function verifyBundleContainsAtLeast(bundle: Bundle, number: number, resourceType: string) {
+    const matchingEntries = getMatchingEntries(bundle, resourceType)
+    if (matchingEntries.length < number) {
+        throw Boom.badRequest(
+            "Bundle must contain at least " + number + " resource(s) of type " + resourceType,
```

Comment:
Interpolated strings with back-ticks could be used help readability e.g. `Bundle entry must contain at least ${number} resource(s) of type ${resourceType}`